lib/python/picongpu/picmi/diagnostics/openpmd_sources/bound_electron_density.py

```diff
@@ -0,0 +1,74 @@
+"""
+This file is part of PIConGPU.
+Copyright 2021-2025 PIConGPU contributors
+Authors: Masoud Afshari
+License: GPLv3+
+"""
+
+import typeguard
+from typing import Optional
+
+
+@typeguard.typechecked
+class BoundElectronDensity:
```

Comment:
this class should inherit from a common interface of all Sources for the openPMD plugin